qt/python/mantidqt/widgets/matrixworkspacedisplay/view.py

```diff
@@ -0,0 +1,171 @@
+        plot_bin_with_errors_action = QAction(self.GRAPH_ICON, "Plot bin (values + errors)", horizontalHeader)
+        plot_bin_with_errors_action.triggered.connect(partial(self.presenter.action_plot_bin_with_errors, table))
+        separator1 = QAction(horizontalHeader)
+        separator1.setSeparator(True)
+
+        horizontalHeader.addAction(copy_bin_values)
+        horizontalHeader.addAction(separator1)
+        horizontalHeader.addAction(plot_bin_action)
+        horizontalHeader.addAction(plot_bin_with_errors_action)
+
+        verticalHeader = table.verticalHeader()
+        verticalHeader.setContextMenuPolicy(Qt.ActionsContextMenu)
+        verticalHeader.setSectionResizeMode(QHeaderView.Fixed)
+
+        copy_spectrum_values = QAction(self.COPY_ICON, "Copy", verticalHeader)
+        copy_spectrum_values.triggered.connect(
+            partial(self.presenter.action_copy_spectrum_values, table, ws_read_function))
+
+        plot_spectrum_action = QAction(self.GRAPH_ICON, "Plot spectrum (values only)", verticalHeader)
+        plot_spectrum_action.triggered.connect(partial(self.presenter.action_plot_spectrum, table))
+        plot_spectrum_with_errors_action = QAction(self.GRAPH_ICON, "Plot spectrum (values + errors)",
+                                                   verticalHeader)
+        plot_spectrum_with_errors_action.triggered.connect(
+            partial(self.presenter.action_plot_spectrum_with_errors, table))
+        separator1 = QAction(verticalHeader)
+        separator1.setSeparator(True)
+
+        verticalHeader.addAction(copy_spectrum_values)
+        verticalHeader.addAction(separator1)
+        verticalHeader.addAction(plot_spectrum_action)
+        verticalHeader.addAction(plot_spectrum_with_errors_action)
+
+    def set_model(self, model_x, model_y, model_e):
+        self._set_table_model(self.table_x, model_x, MatrixWorkspaceTableViewModelType.x)
+        self._set_table_model(self.table_y, model_y, MatrixWorkspaceTableViewModelType.y)
+        self._set_table_model(self.table_e, model_e, MatrixWorkspaceTableViewModelType.e)
+
+    @staticmethod
+    def _set_table_model(table, model, expected_model_type):
+        assert model.type == expected_model_type, \
+            "The model for the table with {0} values has a wrong model type: {1}".format(expected_model_type.upper(),
+                                                                                         model.model_type)
+        table.setModel(model)
+
+    @staticmethod
+    def copy_to_clipboard(data):
+        """
+        Uses the QGuiApplication to copy to the system clipboard.
+
+        :type data: str
+        :param data: The data that will be copied to the clipboard
+        :return:
+        """
+        # This function is likely to be OS specific
+        if not sys.platform == "win32":
+            warning("Copying to clipboard has not been tested on UNIX systems.")
```

Comment:
Works for me on Ubuntu 18.04. Maybe get someone to test OSX then remove this message?

Reply:
@samueljackson92 Would you be able to do a quick test if the copy works? Try to copy a spectrum or a bin's data

Reply:
@DTasev No problem. I'm building your branch now.

Reply:
@DTasev on OSX:
 - Highlighting a single cell and copying (either right click or command+c) works fine
 - Highlighting a whole row/column and copying works fine with right click but not with cmd+c
 - Highlighting a subset (e.g. 3 cells in either across of downwards) only copies the first one. Is this the intended behaviour?

Reply:
Copying a row now doesn't add spaces between the values so you end up with _e.g._
```
1.07.020.033.055.0125.0202.0254.0343.0...
```

Reply:
It adds `\t` now as the original MantidPlot implementation `1779.41899494	1674.0	1627.54748735`, is your editor properly displaying tabs?

Reply:
So it does, where I was pasting the output was not showing them for some reason.

Reply:
This now works fine for me on OSX, but I still get the warning:
> WARNING:root:Copying to clipboard has not been tested on UNIX systems.
when I copy cells. Should this now be removed?

Reply:
I agree with @samueljackson92. Can we remove this warning? I think it might be better not to add OS-specific warnings like this in the code in the future as they could easily be missed in a large code review like this. I would instead make reviewers aware in the test instructions this is something that needs cross-platform testing.